### PR TITLE
[MM-26532] Fix team members modal user count issues

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

The paging logic assumes that whatever page you are on, that you have a full page of users to display, which does not take into account deleted users that have been filtered out client-side. This PR makes it so the frontend requests only active users for each page.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-26532
https://mattermost.atlassian.net/browse/MM-26022

#### Related Pull Requests

https://github.com/mattermost/mattermost-server/pull/14923

### CODE REVIEW

File: components/searchable_user_list/searchable_user_list.jsx [hunk 1]
```diff
@@ -172,10 +172,14 @@ class SearchableUserList extends React.PureComponent {
...
-            const pageEnd = pageStart + this.props.usersPerPage;
+            let pageEnd = pageStart + this.props.usersPerPage;
+            if (this.props.users.length < pageEnd) {
+                pageEnd = this.props.users.length;
+            }
+
             usersToDisplay = this.props.users.slice(pageStart, pageEnd);
 
-            if (pageEnd < this.props.users.length) {
+            if (pageEnd < this.props.total) {
```
Comment (mickmister) [inline components/searchable_user_list/searchable_user_list.jsx:182]: For context on this fix, `this.props.users` is all currently fetched users on this team, with the deleted users filtered out client-side, and `this.props.total` is the total number of non-deleted users on this team, taken from the `stats` API endpoint.

**Comment by amyblais** [comment]
@mickmister Can we assign someone else to review than Harrison as he's off this week?

**Comment by mickmister** [comment]
@jgilliam17 In order to test this, you will need at least 100 users in your test instance. I'm not sure how to easily accomplish this in the instances set up with the `Setup Test Server` label. Here's the steps:

- Have around 200 users on a team (some number above 100)
- Deactivate a user or a few users that have a username towards the beginning of the alphabet (one of the first 100 alphabetically)
- Navigate to a channel where you are the only member
- Refresh the browser
- Go to `Top hamburger menu > Manage Members`

Old code behavior:
- Navigating to Page 2 of the modal will not have a full page of 50 people. Even though it says `51-100` at the top, the deactivated users will be "in the list" but not visible
- Next button will not be visible to go to the next page

New code behavior:
- Page 2 is full of 50 active users
- Next button is visible and clicking goes to next page

Also, this server PR needs to be merged https://github.com/mattermost/mattermost-server/pull/14923 for this to work properly

**Comment by jgilliam17** [comment]
@mickmister Thank you for the test steps.
Yes, I added 50 test users via `/test`, but for some reason can't add more. I may have to destroy the server and start over with a higher number.

**Comment by mickmister** [comment]
@jgilliam17

> Also, this server PR needs to be merged [mattermost/mattermost-server#14923](https://github.com/mattermost/mattermost-server/pull/14923) for this to work properly

This is true for the "Page 2 should have a full page of 50 users" portion, but this webapp PR alone fixes the next button being visible when there are more to load.

**Comment by jgilliam17** [comment]
Note: Added 250 users to this test server, waiting for the other PR to be merge to test.

**Comment by mickmister** [comment]
@jgilliam17 Server PR is merged now

**Comment by Willyfrog** [comment]
tests are probably failing due to https://mattermost.atlassian.net/browse/MM-26512 and https://github.com/mattermost/mattermost-webapp/pull/5797

**Pull request merged**